numpy/ma/core.py

```diff
@@ -2964,8 +2964,6 @@ def __array_finalize__(self, obj):
         # mask), then we make a copy of the template object mask and use
         # that. This is also horribly broken but somewhat less so. Maybe.
         if isinstance(obj, ndarray):
-            # XX: This looks like a bug -- shouldn't it check self.dtype
-            # instead?
```

Comment:
Since you think the code below is fine, can you add a comment as to why? Does this even need to be here at all? If `obj` has fields, then `getmask` and `getmaskarray` are the same thing anyway

Reply:
In fact, I think this comment is correct - if `self` is structured, but `old` is not, then this needs to promote `nomask` into a structured array. So this _does_ need to be `self`

Reply:
I guess we can get here with a structured view of an unstructured array, and vice versa, and both would need to be dealt with. I think you're right that we're better off doing it in a different PR, though, with proper test cases. So, I added the comment back.